viamonvif/device/device.go

```diff
@@ -307,15 +308,20 @@ func (dev Device) callOnvifServiceMethod(endpoint string, method interface{}) ([
 		}
 	}
 
-	return dev.sendSoap(endpoint, soap.String())
+	return dev.sendSoap(ctx, endpoint, soap.String())
 }
 
-func (dev *Device) sendSoap(endpoint, message string) ([]byte, error) {
+func (dev *Device) sendSoap(ctx context.Context, endpoint, message string) ([]byte, error) {
 	contentType := "application/soap+xml; charset=utf-8"
-	//nolint: noctx
-	// TODO(Nick S): This is pretty bad as it can cause the goroutine calling this to hang forever
-	// this should be converted to the http interface that takes a context
-	resp, err := dev.params.HTTPClient.Post(endpoint, contentType, bytes.NewBufferString(message))
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, endpoint, bytes.NewBufferString(message))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+	req.Header.Set("Content-Type", contentType)
+
+	// Using Do instead of POST to support context cancellation and timeout.
+	resp, err := dev.params.HTTPClient.Do(req)
```

Comment:
My understanding i sthat the fix is in these lines - by changing the interface to the one that can take in a context and not hang.
Can you run the new discovery binary against the olde code and verify that we're not missing any of the discovered devices in office?

Reply:
Ran the discovery example client on main vs. feature with some known credentials and both gave the same exact outputs